CI: move to new resource pool and folder in VMC and migrate to new IPAM

chrischdi opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:

CAPV maintainer here.
We currently clean up the CI environment a bit.

For better separation we created a new directory and resource pool which then is used exclusively by cloud-provider-vsphere.

This PR:
* changes the configuration to make use of the new folder and resource pool.
* Makes use of the new resources from ipam in-cluster provider in `hack/e2e.sh` to get ip addresses.

This PR should get cherry-picked to all releases where we still run CI, according test-infra this is all release branches >= v1.22:

- release-1.22
- release-1.23
- release-1.24
- release-1.25
- release-1.26
- release-1.27
- release-1.28

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
NONE
```

chrischdi commented:
FYI @killianmuldoon @sbueringer :-)

chrischdi commented:
> /cherry-pick release-1.28

Cherry-pick bot seems not to be active for this repo. So I'll have to create the PRs manually.

sbueringer commented:
/lgtm

from a VMC point of view.

Is the account that has permissions on this resource pool & folder already in place?

I don't remember merging a PR for it and just wondering why tests are green here / if that means that they now fail on other PRs (but I might have just forgot :D)

chrischdi commented:
> /lgtm
> 
> from a VMC point of view.
> 
> Is the account that has permissions on this resource pool & folder already in place?
> 
> I don't remember merging a PR for it and just wondering why tests are green here / if that means that they now fail on other PRs (but I might have just forgot :D)
> 
> EDIT: Ah we're probably currently using the admin account?

Yes it will still use the admin account.

After merging + cherry-picking this, I will go ahead and open a PR in test-infra to switch over to the non-admin credentials (individual preset for cloud-provider-vsphere which refers a different key for user + password).

chrischdi commented:
/test help

chrischdi commented:
/test pull-cloud-provider-vsphere-verify-shell

chrischdi commented:
/hold

To control when it gets merged as we have to increase the nr of available IP's for the in-cluster ipam provider + decrease the nr of available ips for the metal3 provider.

killianmuldoon reviewed (commented):
/lgtm

lubronzhan commented:
> /hold
> 
> To control when it gets merged as we have to increase the nr of available IP's for the in-cluster ipam provider + decrease the nr of available ips for the metal3 provider.

Should I hold approving this PR?

chrischdi commented:
> > /hold
> > To control when it gets merged as we have to increase the nr of available IP's for the in-cluster ipam provider + decrease the nr of available ips for the metal3 provider.
> 
> Should I hold approving this PR?

Nope, that's fine, I just wanted to unhold during EMEA times when I'm online, so I can also adjust the relevant config in CI (so we don't run out of IPs)

chrischdi commented:
/cherry-pick release-1.28

chrischdi commented:
/cherry-pick release-1.27

chrischdi commented:
/cherry-pick release-1.26

chrischdi commented:
/cherry-pick release-1.26

chrischdi commented:
/cherry-pick release-1.25

chrischdi commented:
/cherry-pick release-1.24

chrischdi commented:
/cherry-pick release-1.23

chrischdi commented:
/cherry-pick release-1.22

lubronzhan commented:
/lgtm
/approve

chrischdi commented:
/hold cancel

chrischdi commented:
> @chrischdi: #763 failed to apply on top of branch "release-1.22":
> 
> ```
> Applying: CI: move to new resource pool and folder in VMC and use in-cluster provider for IPAM
> Using index info to reconstruct a base tree...
> A	hack/e2e.sh
> A	hack/ipclaim-template.yaml
> A	test/e2e/config/vsphere-ci.yaml
> Falling back to patching base and 3-way merge...
> CONFLICT (modify/delete): test/e2e/config/vsphere-ci.yaml deleted in HEAD and modified in CI: move to new resource pool and folder in VMC and use in-cluster provider for IPAM. Version CI: move to new resource pool and folder in VMC and use in-cluster provider for IPAM of test/e2e/config/vsphere-ci.yaml left in tree.
> CONFLICT (modify/delete): hack/ipclaim-template.yaml deleted in HEAD and modified in CI: move to new resource pool and folder in VMC and use in-cluster provider for IPAM. Version CI: move to new resource pool and folder in VMC and use in-cluster provider for IPAM of hack/ipclaim-template.yaml left in tree.
> CONFLICT (modify/delete): hack/e2e.sh deleted in HEAD and modified in CI: move to new resource pool and folder in VMC and use in-cluster provider for IPAM. Version CI: move to new resource pool and folder in VMC and use in-cluster provider for IPAM of hack/e2e.sh left in tree.
> error: Failed to merge in the changes.
> hint: Use 'git am --show-current-patch=diff' to see the failed patch
> Patch failed at 0001 CI: move to new resource pool and folder in VMC and use in-cluster provider for IPAM
> When you have resolved this problem, run "git am --continue".
> If you prefer to skip this patch, run "git am --skip" instead.
> To restore the original branch and stop patching, run "git am --abort".
> ```

v1.22 does not have a `hack/e2e.sh` so tests will never succeed there. Will not cherry-pick it to it.

wyike commented:
/test pull-cloud-provider-vsphere-integration-test
